scan.go

```diff
@@ -33,6 +33,7 @@ type ScanService struct {
 	types        []string
 	keepAlive    string
 	searchSource *SearchSource
+	source       map[string]interface{}
```

Comment:
The `source` should be more flexible. It should be an `interface{}` and we should try our best to make it the body of the request. It should accept at least `io.Reader` and `string` and fall back to serialize as JSON (as we'd do with `map[string]interface{}`).